src/python/pants/backend/python/dependency_inference/import_parser_test.py

```diff
@@ -203,6 +203,10 @@ def test_works_with_python2(rule_runner: RuleRunner) -> None:
         import demo
         from project.demo import Demo
 
+        __import__(b"pkg_resources")
+        __import__(u"pkg_resources")
```

Comment:
These should be different - otherwise we can't be certain that both are being read like we want. Something like this:

```python
        __import__(b"pkg_resources_bytes")
        __import__(u"pkg_resources")
```

Or, remove the first line, as you already test byte strings below. The important thing we're missing is a guarantee that unicode strings are handled properly.

Reply:
Good point. Fixed.